###Objetivo: conferir a estrutura das duas tabelas (colunas e tipos), o número de linhas e uma amostra dos registros.

In [0]:
%sql
-- Colunas e tipos da tabela de leads (Tabela: olist_marketing_qualified_leads_dataset)
DESCRIBE workspace.portfolio_1_olist.olist_marketing_qualified_leads_dataset;

col_name,data_type,comment
mql_id,string,null
first_contact_date,date,null
landing_page_id,string,null
origin,string,null


In [0]:
%sql
-- Colunas e tipos da tabela de negócios fechados (Tabela: olist_closed_deals_dataset)
DESCRIBE portfolio_1_olist.olist_closed_deals_dataset;

col_name,data_type,comment
mql_id,string,null
seller_id,string,null
sdr_id,string,null
sr_id,string,null
won_date,timestamp,null
business_segment,string,null
lead_type,string,null
lead_behaviour_profile,string,null
has_company,boolean,null
has_gtin,boolean,null


In [0]:
%sql
-- Número de linhas: total de leads (Tabela: olist_marketing_qualified_leads_dataset)
SELECT COUNT(*) AS total_leads
FROM workspace.portfolio_1_olist.olist_marketing_qualified_leads_dataset;

total_leads
8000


In [0]:
%sql
-- Número de linhas: total de negócios fechados (Tabela: olist_closed_deals_dataset)
SELECT COUNT(*) AS total_negocios
FROM portfolio_1_olist.olist_closed_deals_dataset;

total_negocios
842


In [0]:
%sql
-- Amostra de 10 leads (Tabela: olist_marketing_qualified_leads_dataset)
SELECT * FROM workspace.portfolio_1_olist.olist_marketing_qualified_leads_dataset LIMIT 10;


mql_id,first_contact_date,landing_page_id,origin
dac32acd4db4c29c230538b72f8dd87d,2018-02-01,88740e65d5d6b056e0cda098e1ea6313,social
8c18d1de7f67e60dbd64e3c07d7e9d5d,2017-10-20,007f9098284a86ee80ddeb25d53e0af8,paid_search
b4bc852d233dfefc5131f593b538befa,2018-03-22,a7982125ff7aa3b2054c6e44f9d28522,organic_search
6be030b81c75970747525b843c1ef4f8,2018-01-22,d45d558f0daeecf3cccdffe3c59684aa,email
5420aad7fec3549a85876ba1c529bd84,2018-02-21,b48ec5f3b04e9068441002a19df93c6c,organic_search
28bdfd5f057764b54c38770f95c69f2f,2018-01-14,22c29808c4f815213303f8933030604c,organic_search
126a0d10becbaafcb2e72ce6848cf32c,2018-05-15,6a110e795dd487f1cf8d7583671987af,email
f76136f54d14a3345951f25b7932366b,2018-05-24,d51b0d02f063ba1d053db6d97226eec3,email
2f838cade4a6012a6cb1016d1d8d95ed,2017-11-10,aeac92c0f5ae22a04ed3b746cce3a1b6,organic_search
7281942387a1a0c3f72a50a8b0bb0920,2017-12-25,88740e65d5d6b056e0cda098e1ea6313,social


In [0]:
%sql
-- Amostra de 10 negócios fechados (Tabela: olist_closed_deals_dataset)
SELECT * FROM portfolio_1_olist.olist_closed_deals_dataset LIMIT 10;

mql_id,seller_id,sdr_id,sr_id,won_date,business_segment,lead_type,lead_behaviour_profile,has_company,has_gtin,average_stock,business_type,declared_product_catalog_size,declared_monthly_revenue
5420aad7fec3549a85876ba1c529bd84,2c43fb513632d29b3b58df74816f1b06,a8387c01a09e99ce014107505b92388c,4ef15afb4b2723d8f3d81e51ec7afefe,2018-02-26T19:58:54.000Z,pet,online_medium,cat,null,null,null,reseller,null,0.0
a555fb36b9368110ede0f043dfc3b9a0,bbb7d7893a450660432ea6652310ebb7,09285259593c61296eef10c734121d5b,d3d1e91a157ea7f90548eef82f1955e3,2018-05-08T20:17:59.000Z,car_accessories,industry,eagle,null,null,null,reseller,null,0.0
327174d3648a2d047e8940d7d15204ca,612170e34b97004b3ba37eae81836b4c,b90f87164b5f8c2cfa5c8572834dbe3f,6565aa9ce3178a5caf6171827af3a9ba,2018-06-05T17:27:23.000Z,home_appliances,online_big,cat,null,null,null,reseller,null,0.0
f5fee8f7da74f4887f5bcae2bafb6dd6,21e1781e36faf92725dde4730a88ca0f,56bf83c4bb35763a51c2baab501b4c67,d3d1e91a157ea7f90548eef82f1955e3,2018-01-17T13:51:03.000Z,food_drink,online_small,null,null,null,null,reseller,null,0.0
ffe640179b554e295c167a2f6be528e0,ed8cb7b190ceb6067227478e48cf8dde,4b339f9567d060bcea4f5136b9f5949e,d3d1e91a157ea7f90548eef82f1955e3,2018-07-03T20:17:45.000Z,home_appliances,industry,wolf,null,null,null,manufacturer,null,0.0
b94fba7670eeb44dce2a0d8eb790e9f5,1c742ac33582852aaf3bcfbf5893abcf,fdb16d3cbbeb5798f2f66c4096be026d,495d4e95a8cf8bbf8b432b612a2aa328,2018-02-07T18:04:05.000Z,health_beauty,online_medium,null,null,null,null,manufacturer,null,0.0
c3e30ed7ac989117c7e1e719b4ac128f,92d7568ad0c5c76fd7d341b2d46f24d6,4b339f9567d060bcea4f5136b9f5949e,85fc447d336637ba1df43e793199fbc8,2018-04-16T18:18:22.000Z,computers,online_medium,wolf,null,null,null,reseller,null,0.0
b02c89251106e1fdd9d92744be9f94f2,44ed138eca6214d572ce1d813fb0049b,34d40cdaf94010a1d05b0d6212f9e909,4ef15afb4b2723d8f3d81e51ec7afefe,2018-04-17T17:01:57.000Z,health_beauty,offline,cat,null,null,null,manufacturer,null,0.0
a90a37898cc5f2718385a2fb981caaff,0b28859cd04d23edefee9c591fb03cd8,f42a2bd194f7802ab052a815c8de65b7,6565aa9ce3178a5caf6171827af3a9ba,2018-05-14T18:37:15.000Z,household_utilities,offline,eagle,null,null,null,reseller,null,0.0
0173e8d8b1d94a355b440fb67388f532,87d73636a3acf123e842bb890a4db036,9d12ef1a7eca3ec58c545c678af7869c,9e4d1098a3b0f5da39b0bc48f9876645,2018-04-24T03:00:00.000Z,household_utilities,online_medium,eagle,null,null,null,manufacturer,null,0.0


### Observações

- A tabela de leads (olist_marketing_qualified_leads_dataset) tem 8000 linhas e 4 colunas, uma linha por lead.
- A tabela de negócios fechados (olist_closed_deals_dataset) tem 842 linhas e 14 colunas, uma linha por venda.
- A coluna `mql_id` aparece nas duas tabelas e é o que liga cada lead ao negócio que ele gerou.
- Os tipos foram reconhecidos corretamente: as datas como `date` e `timestamp`, os campos sim/não como `boolean` e os valores declarados como número. Os cálculos de tempo das próximas perguntas não vão exigir conversão.
- Alguns campos da tabela de negócios (`has_company`, `has_gtin`, `average_stock`, `declared_product_catalog_size`) vêm pouco preenchidos, então têm uso limitado nas análises.